### PR TITLE
add fixes for socket shutdown

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* Fix connection retry attempts for cluster-internal TLS connections that ran into the 15 seconds timeout during the connection establishing attempt. In this case, the low-level socket was repurposed, but not reset properly. This could leave the connection in an improper state and lead to callbacks for some requests to not being called as expected.
  The connection timeout was also increased from 15 seconds to 60 seconds.

  The PR also adds further tests to fuerte.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.12.0: -
  - [ ] Backport for 3.11: https://github.com/arangodb/arangodb/pull/20870
  - [ ] Backport for 3.10: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by mpoeter — APPROVED** [review]

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by neunhoef — APPROVED** [review]
LGTM

File: 3rdParty/fuerte/src/AsioSockets.h [hunk 1]
```diff
@@ -167,10 +234,23 @@ struct Socket<fuerte::SocketType::Ssl> {
...
                                  std::move(done));
+        }, [this]() { 
+          return canceled; 
         });
   }
 
+  void rearm() {
+    // move away old socket, in case it is still in use by some background operation
+    FUERTE_ASSERT(deadSockets.capacity() > deadSockets.size());
+    deadSockets.push_back(std::move(socket));
```
Comment (mpoeter) [inline 3rdParty/fuerte/src/AsioSockets.h:245]: Still needed?

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,14 @@
 devel
 -----
 
+* Fix connection retry attempts for cluster-internal TLS connections that ran
+  into the 15 seconds timeout during the connection establishing attempt.
+  In this case, the low-level socket was repurposed, but not reset properly.
+  This could leave the connection in an improper state and lead to callbacks
+  for some requests to not being called as expected.
+  The fix is to remove the retry logic, and instead prolong the connection
+  timeout to a multiple of the previous timeout value.
```
Comment (mpoeter) [inline CHANGELOG:10]:
```suggestion
```

**Review by mpoeter — APPROVED** [review]

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**